staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/serviceresolver.go

```diff
@@ -14,22 +14,30 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Package webhook checks a webhook for configured operation admission
-package webhook
+package config
 
 import (
 	"errors"
 	"fmt"
 	"net/url"
 )
 
+// ServiceResolver knows how to convert a service reference into an actual location.
+type ServiceResolver interface {
+	ResolveEndpoint(namespace, name string) (*url.URL, error)
+}
+
 type defaultServiceResolver struct{}
 
+func NewDefaultServiceResolver() ServiceResolver {
+	return &defaultServiceResolver{}
+}
+
 // ResolveEndpoint constructs a service URL from a given namespace and name
 // note that the name and namespace are required and by default all created addresses use HTTPS scheme.
 // for example:
 //  name=ross namespace=andromeda resolves to https://ross.andromeda.svc:443
-func (sr defaultServiceResolver) ResolveEndpoint(namespace, name string) (*url.URL, error) {
+func (sr *defaultServiceResolver) ResolveEndpoint(namespace, name string) (*url.URL, error) {
```

Comment:
Why make this a pointer? It has no member variables.

Reply:
Done.